galaxy/api/v2/views/collection_version.py

```diff
@@ -15,21 +15,70 @@
 # You should have received a copy of the Apache License
 # along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
 
-from rest_framework import exceptions as drf_exc
+from django.shortcuts import get_object_or_404
 from rest_framework import views
-from rest_framework import status as http_codes
-from rest_framework.permissions import IsAuthenticated
+from rest_framework import generics
+from rest_framework.response import Response
+from rest_framework.permissions import AllowAny
+
+from galaxy.main import models
+from galaxy.api.v2 import serializers
+from galaxy.api.v2.pagination import CustomPagination
 
 
 __all__ = (
-    'CollectionVersionView',
+    'VersionListView',
+    'VersionDetailView',
 )
 
 
-class CollectionVersionView(views.APIView):
-    permission_classes = (IsAuthenticated, )
+# TODO(awcrosby): Move to views/utils.py and reuse across views
+def _lookup_collection(kwargs):
+    """Helper method to get collection from id, or namespace and name."""
+    pk = kwargs.get('pk', None)
+    ns_name = kwargs.get('namespace', None)
+    name = kwargs.get('name', None)
+
+    if pk:
+        return get_object_or_404(models.Collection, pk=pk)
+
+    ns = get_object_or_404(models.Namespace, name=ns_name)
+    return get_object_or_404(models.Collection, namespace=ns, name=name)
+
+
+class VersionListView(generics.ListAPIView):
+    permission_classes = (AllowAny, )
+    serializer_class = serializers.VersionSummarySerializer
+    pagination_class = CustomPagination
+
+    def get_queryset(self):
+        """Return list of versions for a specific collection."""
+        collection = _lookup_collection(self.kwargs)
+        return models.CollectionVersion.objects.filter(collection=collection)
```

Comment:
@cutwater @alikins To sort a collection version list by semver, we need to evaluate the queryset, but for the drf pagination classes, it takes a queryset (also evaluating a large queryset is not ideal). An option was mentioned earlier to change `CollectionVersion.version` type from `CharField` to `semantic_version.django_fields.VersionField`, this looks like a good solution, thoughts?

Reply:
1. Ok, we may replace `CharField` with `VersionField`, it will make it easier to work with, but doesn't really solve large query evaluation problem, because `VersionField` is just a wrapper for conversion between python code and database.
2. DRF pagination classes able to paginate data or queryset.

Reply:
DRF is now paginating the data, and versions are sorted by semantic version